packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_class_table.dart

```diff
@@ -117,29 +118,85 @@ class _AllocationTracingTableState extends State<AllocationTracingTable> {
-    return DualValueListenableBuilder<bool, List<TracedClass>>(
-      firstListenable: widget.controller.refreshing,
-      secondListenable: widget.controller.classList,
-      builder: (context, _, classList, __) {
-        return FlatTable<TracedClass>(
-          columns: columns,
-          data: classList,
-          keyFactory: (e) => Key(e.cls.id!),
-          onItemSelected: widget.controller.selectTracedClass,
-          sortColumn: _checkboxColumn,
-          secondarySortColumn: secondarySortColumn,
-          sortDirection: sortDirection,
-          selectionNotifier: widget.controller.selectedTracedClass,
-          onSortChanged: (column, direction, {secondarySortColumn}) {
-            // Keep track of sorting state so it doesn't get reset when
-            // `controller.refreshing` changes.
-            setState(() {
-              sortDirection = direction;
-              secondarySortColumn = secondarySortColumn;
-            });
+    return Column(
+      children: [
+        Padding(
+          padding: const EdgeInsets.only(bottom: densePadding),
+          child: ClassFilterTextField(
+            controller: widget.controller,
+          ),
+        ),
+        DualValueListenableBuilder<bool, List<TracedClass>>(
+          firstListenable: widget.controller.refreshing,
+          secondListenable: widget.controller.classList,
+          builder: (context, _, classList, __) {
+            return Expanded(
+              child: OutlineDecoration(
+                child: FlatTable<TracedClass>(
+                  columns: columns,
+                  data: classList,
+                  keyFactory: (e) => Key(e.cls.id!),
+                  onItemSelected: widget.controller.selectTracedClass,
+                  sortColumn: _checkboxColumn,
+                  secondarySortColumn: secondarySortColumn,
+                  sortDirection: sortDirection,
+                  selectionNotifier: widget.controller.selectedTracedClass,
+                  onSortChanged: (column, direction, {secondarySortColumn}) {
+                    // Keep track of sorting state so it doesn't get reset when
+                    // `controller.refreshing` changes.
+                    setState(() {
+                      sortDirection = direction;
+                      secondarySortColumn = secondarySortColumn;
+                    });
+                  },
+                ),
+              ),
+            );
           },
-        );
-      },
+        ),
+      ],
+    );
+  }
+}
+
+class ClassFilterTextField extends StatelessWidget {
+  ClassFilterTextField({
+    Key? key,
+    required this.controller,
+  }) : super(key: key);
+
+  final AllocationProfileTracingViewController controller;
+  final _textEditingController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    return TextField(
+      controller: _textEditingController,
+      decoration: InputDecoration(
+        constraints: const BoxConstraints(
```

Comment:
if you use the clearInputButton helper below I think you can get rid of these constraints

Reply:
Removing these constraints results in an ~50% increase in the text field's height, even when using `clearInputButton`.